### PR TITLE
Update file metadata from provider

### LINKED ISSUE #623: Update file metadata from provider
- [ ] Get the type and the length from the Provider API call
- [ ] Update the UI to reflect the metadata

### PR DESCRIPTION
**Pull request by claudiaHash**
Fixes #623  .

Changes proposed in this PR:

-  file metadata fetched from provider
- UI updated with new data

### CODE REVIEW

File: src/components/atoms/File.tsx
Comment (kremalicious) [inline src/components/atoms/File.tsx:45]: the file icon should be rendered at all times so there will be no layout shift. Only its content should load and be replaced

**Comment by kremalicious** [comment]
were you able to make this actually work, specifically the file size on some asset? Cause we run into issues over in #479 where `provider` seems to never return any `contentLength`.

**Comment by claudiaHash** [comment]
For the great majority of the assets, I could, yes. But there are some assets like: `did:op:555a0DD8B24963e38FFF1a0c84F13f694Bd0f4F6` , `did:op:d5766cc73AE44977b4F7d72cDd3b33740101e830`, `did:op:1D9a0Ad91a5bD3d4091fD474c481A3079A846260` for which I've got a  `index: 0; valid: false` response although on live there is valid data for them. 
However, I ran  into some issues when I sent the did as a string, not as a DID.

**Comment by krisliew** [comment]
> However, I ran into some issues when I sent the did as a string, not as a DID.

@claudiaHash When passing string argument, it can only accept file endpoint url for `getFileInfo()` 
See here too: https://github.com/oceanprotocol/provider/blob/79de2a4aabd248ac2c72184d7067c58ac3ca7bbf/ocean_provider/routes/consume.py#L149

File: src/components/organisms/AssetActions/index.tsx
Comment (KatunaNorbert) [inline src/components/organisms/AssetActions/index.tsx:23]: Looks like there should be a default value in here otherwise due to this `if (!file) return null`  check from the `File` component the component will not show the content. That's why you don't get the component. 
Might be set to an empty object or create an object with the data from the `ddo` until you get the data from provider

Comment (claudiaHash) [reply src/components/organisms/AssetActions/index.tsx:23]: Thought that the problem was elsewhere. Thanks, I'll fix it.

**Comment by claudiaHash** [comment]
> > However, I ran into some issues when I sent the did as a string, not as a DID.
> 
> @claudiaHash When passing string argument, it can only accept file endpoint url for `getFileInfo()`
> See here too: https://github.com/oceanprotocol/provider/blob/79de2a4aabd248ac2c72184d7067c58ac3ca7bbf/ocean_provider/routes/consume.py#L149

Sorry, I don't think I expressed myself correctly. There are some assets without an url in their ddo ( like this one: did:op:6c5Ec27A1e6F14Ea45736fDF115D9b9279aF617a ) so, if we don't send the asset did as DID, we can't get a valid file info data.

**Comment by krisliew** [comment]
> > > However, I ran into some issues when I sent the did as a string, not as a DID.
> > 
> > 
> > @claudiaHash When passing string argument, it can only accept file endpoint url for `getFileInfo()`
> > See here too: https://github.com/oceanprotocol/provider/blob/79de2a4aabd248ac2c72184d7067c58ac3ca7bbf/ocean_provider/routes/consume.py#L149
> 
> Sorry, I don't think I expressed myself correctly. There are some assets without an url in their ddo ( like this one: did:op:6c5Ec27A1e6F14Ea45736fDF115D9b9279aF617a ) so, if we don't send the asset did as DID, we can't get a valid file info data.

Yes, we definitely have to use the asset did (as `DID`) based on `getFileInfo()` implementation. 

For passing `string` param, it is used for `fileinfo()` method to check publish form file URL if i am not mistaken, see here at my PR
https://github.com/oceanprotocol/market/blob/8bc4665a864aff3deeab8adc7e0483699797426e/src/utils/provider.ts#L28-L34

**Comment by mihaisc** [comment]
For the most assets it's ok. 
There are still some issue on some (not related to this pr, these are provider issues, but are reflected on the market) ; (pr vs main)

image
image

image
image

Not sure if we should merge until we make sure the provider information is 100%.

**Comment by claudiaHash** [comment]
> Not sure if we should merge until we make sure the provider information is 100%.

If the structure of the data that comes from provider won't be changed, I think it could be merged. But maybe is safer to wait until the provider info is 100% reliable and merge it then.

**Review by mihaisc — APPROVED** [review]

**Pull request merged**
